src/mpid/ch4/netmod/ofi/ofi_win.h

```diff
@@ -1062,6 +1068,8 @@ static inline int MPIDI_NM_mpi_win_allocate_shared(MPI_Aint size,
         shm_unlink(shm_key);
 
   fn_exit:
+    if (shm_key != NULL)
```

Comment:
This check isn't necessary. The `(re)alloc` with either return a valid address of `NULL`, both of which are safe to pass to `free`. In fact, `NULL` initializing the ptr is not strictly necessary either.